network/client.go

```diff
@@ -307,8 +307,18 @@ func (c *PeerClient) SetWriteDeadline(t time.Time) error {
 	return nil
 }
 
-// IncomingReady returns true if the client has both incoming and outgoing sockets established.
-func (c *PeerClient) IncomingReady() bool {
+// SetIncomingReady sets a client state to ready for the incomming requests.
```

Comment:
It may be better to keep `SetIncomingReady` and `SetOutgoingReady` private, or otherwise double check if the channels have already been closed since developers may incorrectly use the API.

Reply:
done